video/out/wayland_common.c

```diff
@@ -1298,15 +1300,15 @@ int vo_wayland_reconfig(struct vo *vo)
 
     wl->aspect_ratio = vo->dwidth / (float)vo->dheight;
 
-    if (vo->opts->fullscreen) {
+    if (wl->vo_opts->fullscreen) {
         /* If already fullscreen, fix resolution for the frame size change */
-        if (wl->fullscreen && wl->current_output) {
+        if (wl->current_output) {
             wl->geometry.x0  = 0;
             wl->geometry.y0  = 0;
             wl->geometry.x1  = mp_rect_w(wl->current_output->geometry)/wl->scaling;
             wl->geometry.y1  = mp_rect_h(wl->current_output->geometry)/wl->scaling;
         } else {
-            if (vo->opts->fsscreen_id < 0) {
+            if (wl->vo_opts->fsscreen_id < 0) {
                 xdg_toplevel_set_fullscreen(wl->xdg_toplevel, NULL);
             } else {
                 xdg_toplevel_set_fullscreen(wl->xdg_toplevel, wl_out);
```

Comment:
I think this can be simplified a lot now. The `set_fullscreen` will happen in response to the option being handled, so it will never get triggered here.

Reply:
Hmm, the ``--fullscreen`` option breaks with this. This was the part where `set_fullscreen` gets executed  initially.

Reply:
The else doesn't need to be there anymore since `wl->vo_opts` keeps track of the state. The only thing is that the cursor is initially shown on a fullscreen.